### PR TITLE
make dialog background color configurable (#7320)

### PR DESCRIPTION
**Pull request by lequem**
Problem with my previous branch so this new branch to allow the configuration of the dialog's background color.

### CODE REVIEW

**Comment by lequem** [comment]
travis keep failing.

**Comment by abarth** [comment]
LGTM

**Comment by abarth** [comment]
Maybe Travis is flaky...

**Comment by abarth** [comment]
Thanks!

File: packages/flutter/lib/src/material/dialog.dart [hunk 1]
```diff
@@ -37,15 +37,7 @@ class Dialog extends StatelessWidget {
...
-    Brightness brightness = Theme.of(context).brightness;
-    assert(brightness != null);
-    switch (brightness) {
-      case Brightness.light:
-        return Colors.white;
-      case Brightness.dark:
-        return Colors.grey[800];
-    }
-    return null;
+    return Theme.of(context).dialogBackgroundColor;
```
Comment (Hixie) [inline packages/flutter/lib/src/material/dialog.dart]: We should probably just inline this private where it's used.

File: packages/flutter/test/material/dialog_test.dart [hunk 2]
```diff
@@ -55,4 +55,45 @@ void main() {
...
+      )
+    );
+
+    await tester.tap(find.text('X'));
+    await tester.pump(); // start animation
+    await tester.pump(const Duration(seconds: 1));
+
+    StatefulElement widget = tester.element(find.byType(Material).last);
+    Material materialconfig = widget.state.config;
+    //first and second expect check that the material is the dialog's one
```
Comment (Hixie) [inline packages/flutter/test/material/dialog_test.dart]: nit: comments should have a space after the `//`, a capital letter to start, and a period to end.

**Pull request merged**